### PR TITLE
Eng 467 ai xplain sdk update finetune functional tests to cover all new finetunable models

### PR DESCRIPTION
**Pull request by xainaz**
Added createdAt attribute to model class, and implemented necessary changes in Model Factory. 
As for the finetune functional test, I replaced the changes with the original code that ran the test on the 3 models in data, since there were no new models in the past 3 weeks. 
2 models are passing ( llama2 7b, vectara) and 1 is failing : air

### CODE REVIEW

**Comment by thiago-aixplain** [comment]
Where are the functional tests?

File: aixplain/modules/model/__init__.py [hunk 1]
```diff
@@ -84,6 +85,7 @@ def __init__(
         self.backend_url = config.BACKEND_URL
         self.function = function
         self.is_subscribed = is_subscribed
+        self.createdAt = createdAt
 
     def to_dict(self) -> Dict:
         """Get the model info as a Dictionary
```
Comment (thiago-aixplain) [inline aixplain/modules/model/__init__.py:91]: Please check whether this `to_dict` method works with a datetime attribute.

File: tests/functional/finetune/finetune_functional_test.py [hunk 2]
```diff
@@ -95,6 +109,13 @@ def test_end2end(run_input_map):
         assert len(result["details"][0]["metadata"]) > 0
     finetune_model.delete()
 
+def test_end2end(run_input_map, finetunable_llms_from_last_4_weeks):
+    if finetunable_llms_from_last_4_weeks:
+        for model in finetunable_llms_from_last_4_weeks:
+            run_finetune_test(model, run_input_map)
```
Comment (thiago-aixplain) [inline tests/functional/finetune/finetune_functional_test.py:115]: You should not test all the models within the same method. You should call `test_end2end` for each model in `finetunable_llms_from_last_4_weeks`. Check how to use a fixture together with the python command `yield`. There is an example here: https://github.com/aixplain/aiXplain/blob/main/tests/functional/pipelines/designer_test.py#L17

**Review by thiago-aixplain — CHANGES_REQUESTED** [review]

File: aixplain/modules/model/__init__.py (hunk 1 above)
Comment (xainaz) [reply aixplain/modules/model/__init__.py:91]: It seems to work, this is what it returns:
{'id': '64e615671567f848804985e1', 'name': 'GPT2', 'supplier': <Supplier.OPENAI: {'id': 1777, 'name': 'OpenAI', 'code': 'openai'}>, 'additional_info': {'parameters': {'language': ['en']}}, 'createdAt': datetime.datetime(2023, 8, 30, 10, 53, 10, 827000, tzinfo=datetime.timezone.utc)}
I added createdAt to to_dict

**Review by thiago-aixplain — APPROVED** [review]

**Pull request merged**